app/calculation/score-engine.js

```diff
@@ -0,0 +1,238 @@
+
+const { first } = require('lodash')
+const bandHigh = 'High'
+const bandLow = 'Low'
+const bandMedium = 'Medium'
+
+class ScoreEngine {
+  constructor (desirabilityAssessment) {
+    this.scoringData = require('./score-data.json')// score-data.json
+    this.desirabilityAssessment = desirabilityAssessment// grant-scheme.json
+  }
+
+  getScore () {
+    // calculate each question result
+    this.desirabilityAssessment.desirability.questions
+      .map((qanswer, index, allQanswers) => calculate(qanswer, this.scoringData.desirability, allQanswers))
+    const maxScore =
+      this.scoringData.desirability.questions
+        .reduce((total, question) => question.maxScore + total, 0)
+    const noShowResultQuestions =
+      this.scoringData.desirability.questions
+        .filter(question => question.showResult === false)
+    const actualScore = this.desirabilityAssessment.desirability.questions
+      .filter(question =>
+        noShowResultQuestions
+          .every(noShow => noShow.key !== question.key))
+      .reduce(
+        (total, question) =>
+          question.rating.score + total, 0)
+    this.desirabilityAssessment.desirability.overallRating.score = actualScore
+    const bandScore = (actualScore / maxScore) * 100
+
+    // remove noShowResult questions
+    this.desirabilityAssessment.desirability.questions = this.desirabilityAssessment.desirability.questions.filter(question => noShowResultQuestions.every(noShow => noShow.key !== question.key))
+
+    // get overall rating band
+    this.desirabilityAssessment.desirability.overallRating.band = GetOverAllRatingBand(bandScore, this.scoringData.desirability)
+    return this.desirabilityAssessment
+  }
+}
+
+function GetOverAllRatingBand (bandScore, sectionScoringData) {
```

Comment:
lowercase beginning of function name 

GetOverAllRatingBand -> getOverAllRatingBand

Reply:
updated.